scripts/update_changelog_and_credits.py

```diff
@@ -131,10 +132,35 @@ def check_ordering_of_sections(release_summary_lines):
                     section.strip(), next_section.strip()))
 
 
-def update_changelog(release_summary_lines, current_release_version):
+def get_previous_release_version(current_release_version):
+    """Finds previous version given the current version.
+
+    Args:
+        current_release_version: str. The current release version.
+
+    Returns:
+        str. The previous version.
+
+    Raises:
+        Exception. The version format is invalid or there is no previous
+            version for the given version.
+    """
+    all_tags = subprocess.check_output(['git', 'tag'])[:-1].split('\n')
+    # Tags are of format vX.Y.Z. So, the substring starting from index 1 is the
+    # version.
+    previous_release_version = all_tags[-1][1:]
+    # This is for hotfixes.
+    if previous_release_version == current_release_version:
```

Comment:
Just to be safe, let's pass a branch_type parameter to this function, and do this only if it is a hotfix branch. and the above when it is a release branch. In other cases raise an exception. 

I am thinking that somehow if you rerun this function on a release branch, which has already been deployed with the same release number, it might theoretically run this.

Reply:
Done.